### PR TITLE
feat: Project Lifecycle

### LINKED ISSUE #286: Configure project storage location
Now that we have a configuration screen, let's add the ability to configure project storage locations.

### PR DESCRIPTION
**Pull request by cguedes**
Closes #286, #288 

This PR adds four new operations to Ref Studio to control the project lifecycle.

- New Project
- Open Project
- Close Project
- Try Sample Project

| File Menu Entries | Explorer Actions when no project is open |
|--------|--------|
| image | image | 

### New Project

When a new project is created (default directory is the users desktop folder), a modal asks for the project name, and the project is **created empty** (no files, no uploads folder) and with a **unsaved file open**.

**Notes**
  * If the project folder exists, a warning is shown, and all the content is removed.
  * The current project (and open editores) is closed

| Project Name Dialog | Warning message | New Project |
|--------|--------|-------|
Screenshot 2023-08-15 at 15 16 04 | Screenshot 2023-08-15 at 15 17 26 | Screenshot 2023-08-15 at 15 20 09

### Open Project

When a project is requested to be open, a folder explorer modal is opened (opens desktop by default). After selection, the current open project is closed and the new one is open.

**Notes:**
  * The project name show in the explorer is derived from the path (last segment)
  * Any directory can be open

| Select Folder Dialog |
|--------|
| Screenshot 2023-08-15 at 15 29 06 |

### Close project

Closing a project can be done via `File -> Close Project` or via the explorer panel icon (X).
When a project is closed, all the open editors are closed and the project path reset to an empty string "".

### Try Sample Project 

Trying a sample project will create a sample folder `RefStudio Sample` in the Desktop folder with 3 files (the ones we've had before).

**Notes:**
  * If the action if performed again, will remove the sample folder and create a fresh sample project.
  * For now, no PDF (or reference ingestion) is performed for the sample. We should improve this later.

## Extra Notes

* Deprecated `settings.general` keys and created new key for the current open project `project.currentDir`.
   * Code migrate `settings.general.appDataDir + settings.appDataDir.projectName` to the new `project.currentDir` on the first run.
   * The settings file is kept in the application config folder (`/Users/USER/Library/Application Support/studio.ref.desktop/refstudio-settings.json`).
* Updated settings panel -> general to remove project directory.
* Project directory is shown in the footer (bottom left).
* The `uploads` folder creation is delayed to the first upload (so that new projects won't have that empty folder listed).
* Every time a project is created or open, the path is saved in the settings in order to reopen the project automatically on the next run.

### CODE REVIEW

File: src/atoms/projectState.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+// Internal Atoms
+// #####################################################################################
+const currentProjectPathAtom = atom('');
+
+// #####################################################################################
+// Public API
+// #####################################################################################
+export const isProjectOpenAtom = atom((get) => get(currentProjectPathAtom) !== '');
+export const projectPathAtom: Atom<string> = currentProjectPathAtom;
+export const projectNameAtom = selectAtom(currentProjectPathAtom, extractProjectName);
```
Comment (sehyod) [inline src/atoms/projectState.ts:19]: nit: I don't think we need `selectAtom` here, we can use a basic derived atom like this `atom((get) => extractProjectName(get(currentProjectPathAtom)));`

File: src/atoms/referencesState.ts [hunk 2]
```diff
@@ -40,15 +44,16 @@ export const referencesSyncInProgressAtom = atom<boolean>(false);
 // #####################################################################################
 // References initialization
 // #####################################################################################
-const referencesLoadedAtom = atom<boolean>(false);
+const referencesLoadedAtom = atom<boolean | null>(null); // Starts true to account for
```
Comment (sehyod) [inline src/atoms/referencesState.ts:47]: I think the comment is outdated

File: src/io/filesystem.ts [hunk 3]
```diff
@@ -77,11 +85,20 @@ export async function getSystemPath(rsPath: string) {
...
-  const baseDir = await getProjectBaseDir();
+export function getRefStudioPath(absolutePath: string) {
+  const baseDir = getProjectBaseDir();
   return absolutePath.replace(new RegExp(`^${baseDir}`), '');
 }
 
+/**
+ * Returns the the OS-specific separator
+ *
+ * @returns the separator (/ or \)
```
Comment (sehyod) [inline src/io/filesystem.ts:96]:
```suggestion
 * @returns the separator (/ or \\)
```
nit: escape the backslash so it properly shows up in tooltip

File: src/atoms/projectState.ts (hunk 1 above)
Comment (cguedes) [reply src/atoms/projectState.ts:19]: I think it's ok and more legible.

**Review by sehyod — APPROVED** [review]

**Pull request merged**